transcendence-app/src/user/user.controller.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+import { RouteGuard } from '../shared/guards/route.guard';
+import { UserDto } from './dto/user.dto';
+import { UserService } from './user.service';
+
+@Controller('user')
+export class UserController {
+  constructor(private readonly userService: UserService) {}
+
+  @Get(':username')
+  findOne(@Param('username') username: string) {
+    return this.userService.findOne(username);
+  }
+
+  @Get()
+  @UseGuards(RouteGuard)
+  findAll() {
+    return this.userService.findAll();
+  }
+
```

Comment:
We decided to put 'provider' as a field in the UserDto. Why did you decide we don't want it anymore? I can think of some reasons, but I'd like to know your opinion.